pkg/controllers/servicescale/scaler.go

```diff
@@ -305,11 +292,32 @@ func (s *SimpleScale) scrape() error {
 	}
 	stat.readyPods = readyPods
 
-	logrus.Debugf("collect metric for %s/%s, total request: %v, average in-flight request per pod: %v, ready pod: %v", s.namespace, s.serviceName, totalActiveRequest, stat.activeRequest, readyPods)
+	logrus.Debugf("collect metric for %s/%s, total request: %v(inbound %v, outbound %v), average in-flight request per pod: %v, ready pod: %v", s.namespace, s.serviceName, totalActiveRequest, inboundActiveRequests, outbountActiveRequests, stat.activeRequest, readyPods)
 	s.metrics.stats = append(s.metrics.stats, stat)
 	return nil
 }
 
+func calculateActiveRequests(scanner *bufio.Scanner, requestMatch, responseMatch string, direction string) int {
+	var request, response int
+	for scanner.Scan() {
+		if strings.Contains(scanner.Text(), responseMatch) && strings.Contains(scanner.Text(), direction) {
+			parts := strings.Split(scanner.Text(), " ")
+			if len(parts) == 2 {
+				rqs, _ := strconv.Atoi(parts[1])
+				response = rqs
+			}
+		}
+		if strings.Contains(scanner.Text(), requestMatch) && strings.Contains(scanner.Text(), direction) {
+			parts := strings.Split(scanner.Text(), " ")
+			if len(parts) == 2 {
+				rqs, _ := strconv.Atoi(parts[1])
+				request = rqs
+			}
+		}
+	}
+	return request - response
```

Comment:
Is there any way this could be negative ?

Reply:
don't think so, unless linkerd has it wrong.

Reply:
Yea an edge case, I'm just wondering if it blips and then records a bunch of closing requests (not sure if thats theoretically possible). If it does that could screw with autoscaling. Could add an absolute call but that is probably premature optimization. We should just keep in mind.